### PR TITLE
feat: shopware 6.6 compatibility

### PR DESCRIPTION
**Pull request by stefanpoensgen**

### CODE REVIEW

File: src/Content/MailArchive/MailArchiveAttachmentCollection.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+declare(strict_types=1);
+
+namespace Frosh\MailArchive\Content\MailArchive;
+
+use Shopware\Core\Framework\DataAbstractionLayer\EntityCollection;
+
+/**
+ * @extends EntityCollection<MailArchiveAttachmentEntity>
+ */
+class MailArchiveAttachmentCollection extends EntityCollection
```
Comment (shyim) [inline src/Content/MailArchive/MailArchiveAttachmentCollection.php:12]: I would prefer not to have a Collection class when it does not have any method. We can just type `EntityCollection<MailArchiveAttachmentEntity>` and PHPStan is happy

File: src/Content/MailArchive/MailArchiveEntity.php [hunk 2]
```diff
@@ -26,11 +27,6 @@ class MailArchiveEntity extends Entity
 
     protected ?string $transportState;
```
Comment (shyim) [inline src/Content/MailArchive/MailArchiveEntity.php:28]: as you want to release a 3.0.0. We should maybe get rid of all methods here and use public properties instead.

File: src/Controller/Api/MailArchiveController.php [hunk 3]
```diff
@@ -90,9 +97,6 @@ public function download(Request $request): JsonResponse
             throw MailArchiveException::notFound();
         }
 
-        // for backward compatibility
-        $content = $mailArchive->getEml();
```
Comment (shyim) [inline src/Controller/Api/MailArchiveController.php:94]: create $content as empty variable or check for eml path and throw an error when it is not found 🤔. Feels weird to define $content only in a if and use it outside of the if

File: src/Content/MailArchive/MailArchiveAttachmentCollection.php (hunk 1 above)
Comment (tinect) [reply src/Content/MailArchive/MailArchiveAttachmentCollection.php:12]: I like dedicated Collections more than magic typehints, like `EntityRepository<EntityCollection<MailArchiveAttachmentEntity>>`

File: src/Content/MailArchive/MailArchiveEntity.php (hunk 2 above)
Comment (tinect) [reply src/Content/MailArchive/MailArchiveEntity.php:28]: `dal:validate` says: NO

File: src/Content/MailArchive/MailArchiveAttachmentCollection.php (hunk 1 above)
Comment (shyim) [reply src/Content/MailArchive/MailArchiveAttachmentCollection.php:12]: One useless file more in opcache and to autoload. I am fine when we want to keep it

**Review by tinect — APPROVED** [review]

**Review by shyim — APPROVED** [review]

**Pull request merged**